### PR TITLE
fixed the stale issues which fixes issue #282

### LINKED ISSUE #282: Stale issues doesn't repoint at other respositories
When visiting https://code.publiclab.org/#r=mapknitter (using the project selector dropdown), the stale issues listing doesn't seem to load from `mapknitter` but instead the default `plots2`. We should rewire it to point at the appropriate repository. 

We'd love help with this one!

### PR DESCRIPTION
**Pull request by DevanshCodes**
fixes #282 
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #282 `-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

File: spec/staleIssuesUtil/staleIssuesUtil.spec.js [hunk 1]
```diff
@@ -59,4 +68,5 @@ describe('staleIssuesUtil.js', () => {
 	})
 
 
+
```
Comment (somenath1435) [inline spec/staleIssuesUtil/staleIssuesUtil.spec.js]: remove unnecessary space

File: src/utils/staleIssuesUtil.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 let model_utils = require('../models/utils')
 
+
```
Comment (somenath1435) [inline src/utils/staleIssuesUtil.js]: unnecessary

**Comment by DevanshCodes** [comment]
@jywarren @Rishabh570 Can you please review the PR?

**Comment by jywarren** [comment]
Hi, this is great. Love the test fixes too. Can you upload a gif of this working to confirm for us? Thank you!

**Comment by DevanshCodes** [comment]
> Hi, this is great. Love the test fixes too. Can you upload a gif of this working to confirm for us? Thank you!

Hey! My Index.html was not showing the fetched data from API, so I used `console.log(issueArray)` to test from the fake data. Attaching the gif here. As can be seen from the gif, the issues which have only plots 2 repo url is logged in console. 

stale issues

File: src/utils/staleIssuesUtil.js [hunk 3]
```diff
@@ -75,11 +74,35 @@ function getStaleIssues(org, repo) {
...
+    })
+}
 
 
 
 module.exports = {
     getOrgWideIssues: getOrgWideIssues,
-    getStaleIssues: getStaleIssues
+    getStaleIssues: getStaleIssues,
+    getRepoStaleIssues: getRepoStaleIssues
```
Comment (jywarren) [inline src/utils/staleIssuesUtil.js]: This is great. Do we have to hook up this new function to the UI?

Comment (DevanshCodes) [reply src/utils/staleIssuesUtil.js]: Yes, we have to hook this up to the UI, We can just make changes to the community-toolbox.js and in place of calling getStaleIssues function we can call getRepoStaleIssues, which will return the stale issues corresponding to repo selected in drop down.

File: src/utils/staleIssuesUtil.js [hunk 4]
```diff
@@ -75,11 +74,35 @@ function getStaleIssues(org, repo) {
...
+    .then((issues) => {
+        if(issues!=undefined && issues!=null) {
+            return issues;
+        }
+    }).then((issues)=>{
+            issues.map(function mappingToIssues(issue, index) {
+                let str = issue.repository_url;
+                let n = str.search(repo);
+                if(n!=-1 & n!=undefined) {
+                    if(!issueArray.includes(issue)){
```
[the remark this answers is not in the document]
Comment (DevanshCodes) [reply src/utils/staleIssuesUtil.js]: I tried it without including `if(!issueArray.includes(issue)` and then my console.log shows the same issue twice or thrice (I don't remember the exact number). But this if statement fixed the duplicated issues, and now the output contained only issue once.

File: src/utils/staleIssuesUtil.js [hunk 5]
```diff
@@ -75,11 +74,35 @@ function getStaleIssues(org, repo) {
...
+                        issueArray.push(issue);
+                    }
+                }
+            })
+            model_utils.setItem(repo,issueArray);
```
[the remark this answers is not in the document]
Comment (DevanshCodes) [reply src/utils/staleIssuesUtil.js]: Cool, I'll fix that and plug this function in `showStaleIssues` in `community-toolbox.js`.

**Comment by DevanshCodes** [comment]
Hey! @Rishabh570 I plugged the function and fixed the key name. Please take a look.

**Comment by Rishabh570** [comment]
Looks good, can you share a GIF showing stale issues list getting updated according to a particular repository. It's just to make sure that it is working as intended.

**Comment by DevanshCodes** [comment]
> Looks good, can you share a GIF showing stale issues list getting updated according to a particular repository. It's just to make sure that it is working as intended.

Hey, I shared a gif earlier showing the stale issues are showing only for a particular repo using `console.log` but I can't test that in browser, because data is not showing up there. Can you help me with that?

**Comment by Rishabh570** [comment]
@DevanshCodes What does the console say when you try to load it in the browser? Is there some error fetching the stale issues' list.
Make sure you've deleted the `stale issues` key from the DB and build the file. Then try the hard refresh in the browser...if it's showing some error, please share that here (you can insert console logs in between the code to make sure the right functions are getting invoked).

**Comment by DevanshCodes** [comment]
> @DevanshCodes What does the console say when you try to load it in the browser? Is there some error fetching the stale issues' list.
> Make sure you've deleted the `stale issues` key from the DB and build the file. Then try the hard refresh in the browser...if it's showing some error, please share that here (you can insert console logs in between the code to make sure the right functions are getting invoked).

Hey, when I am running the Grunt command, console is stuck on Running 'watch' task  Waiting....

**Comment by Rishabh570** [comment]
Yes, if you just run grunt, it will keep looking for changes in the code
and automatically build it...you can just do "grunt build" so that it justs
builds one time and exits.

**Comment by DevanshCodes** [comment]
Hey, I fixed the data not loading in browser issue, and pushed the changes. Here's the GIF showing that everything is working as it should. Sorry about the GRUNT issue, haven't worked with GRUNT much, though now its clear.

community-toolbox

**Comment by DevanshCodes** [comment]
> Hey, I fixed the data not loading in browser issue, and pushed the changes. Here's the GIF showing that everything is working as it should. Sorry about the GRUNT issue, haven't worked with GRUNT much, though now its clear.
> 
> community-toolbox

Apparently there's no such function as DisplayRepoIssues in community-toolbox.js but was returning it in the same. Hence, was creating the error in browser console, commenting out that line fixed the problem.

**Comment by Rishabh570** [comment]
Looks great! Btw the dropdown and toggle button styles are affected I guess...can you take a look? Thanks.

**Comment by DevanshCodes** [comment]
Fixed the dropdown and toggle button styling issue.

ezgif com-video-to-gif (1)

**Pull request merged**